app/events/domains/proxy_domains_changed_event.rb

```diff
@@ -0,0 +1,29 @@
+require 'uri'
+
+class Domains::ProxyDomainsChangedEvent < BaseEventStoreEvent
+  def self.create(proxy)
+    new(
+      proxy:    proxy,
+      staging_domains: extract_domain(proxy.sandbox_endpoint),
+      production_domains: extract_domain(proxy.endpoint),
+
+      metadata: {
+        provider_id: (provider_id = proxy.provider&.id),
+        zync: {
+          tenant_id: proxy.tenant_id || proxy.provider&.tenant_id || provider_id,
```

Comment:
if u r fetching it in memory in https://github.com/3scale/porta/pull/844/files#diff-5db06a47413cddcb71e8e30a355849cdR11, you should use the same variable here.